services/gundeck/test/integration/API.hs

```diff
@@ -828,63 +848,67 @@ testLongPushToken g _ b _ = do
 
 -- * Helpers
 
-registerUser :: HasCallStack => Gundeck -> Cannon -> Http (UserId, ConnId)
-registerUser gu ca = do
+registerUser :: HasCallStack => TestM (UserId, ConnId)
+registerUser = do
+    gu <- view tsGundeck
+    ca <- view tsCannon
     uid <- randomId
     con <- randomConnId
-    void $ connectUser gu ca uid con
-    ensurePresent gu uid 1
+    void $ connectUser ca uid con
+    ensurePresent uid 1
     return (uid, con)
 
-ensurePresent :: HasCallStack => Gundeck -> UserId -> Int -> Http ()
-ensurePresent gu u n =
+ensurePresent :: HasCallStack => UserId -> Int -> TestM ()
+ensurePresent u n = do
+    gu <- view tsGundeck
     retryWhile ((n /=) . length . decodePresence) (getPresence gu (showUser u)) !!!
         (const n === length . decodePresence)
 
-connectUser :: HasCallStack => Gundeck -> Cannon -> UserId -> ConnId -> Http (TChan ByteString)
-connectUser gu ca uid con = do
-    [(_, [ch])] <- connectUsersAndDevices gu ca [(uid, [con])]
+connectUser :: HasCallStack => CannonR -> UserId -> ConnId -> TestM (TChan ByteString)
+connectUser ca uid con = do
+    [(_, [ch])] <- connectUsersAndDevices ca [(uid, [con])]
     return ch
 
 connectUsersAndDevices
   :: HasCallStack
-  => Gundeck -> Cannon -> [(UserId, [ConnId])]
-  -> Http [(UserId, [TChan ByteString])]
-connectUsersAndDevices gu ca uidsAndConnIds =
-    strip <$> connectUsersAndDevicesWithSendingClients gu ca uidsAndConnIds
+  => CannonR -> [(UserId, [ConnId])]
```

Comment:
Not sure how intense you wanted to get with this refactor; but a lot of these params are still being passed explicitly when they're available in TestM.

It should probably happen in a different PR; but it would be nice to finish the refactoring 😄

Reply:
note that there is two cannons, and in some places the helper doesn't know which one to pick.  could have been done with a bool rather than passing the cannon explicitly, yes.

...  in fact, i want to do that now.

also, i don't like the linter function any more.  how do i make it shut up again?  i'll see if i can find it.